src/lib/chat/matrix-client.ts

```diff
@@ -341,6 +342,16 @@ export class MatrixClient implements IChatClient {
       name = roomNameEvent.getContent().name;
     }
 
+    let lastMessageEvent = null;
+    const timelineEvents = room.getLiveTimeline().getEvents();
```

Comment:
Do you know how many events this actually is? What happens if there are thousands of messages?

Reply:
Was just thinking the same. This could be a massive amount. I'll have a look at the suggestion above re using `messages` property of the conversation.